grakn-graql/src/main/java/ai/grakn/graql/internal/query/analytics/CountQueryImpl.java

```diff
@@ -49,13 +54,38 @@ public Long execute() {
             return 0L;
         }
 
-        ComputerResult result = getGraphComputer().compute(new CountMapReduce(
-                subLabels.stream().map(graph.get().admin()::convertToId).collect(Collectors.toSet())));
-        Map<Serializable, Long> count = result.memory().get(CountMapReduce.class.getName());
+        Set<LabelId> rolePlayerLabelIds = subTypes.stream()
+                .filter(Concept::isRelationType)
+                .map(relationType -> ((RelationType) relationType).relates())
+                .filter(roles -> roles.size() == 2)
+                .flatMap(roles -> roles.stream().flatMap(role -> role.playedByTypes().stream()))
+                .map(type -> graph.get().admin().convertToId(type.getLabel()))
+                .filter(LabelId::isValid)
+                .collect(Collectors.toSet());
```

Comment:
Now we have this same thing in 3 places. @fppt maybe we need some common way to access it? Especially since it's easy to get wrong.